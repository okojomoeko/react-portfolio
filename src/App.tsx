import React from "react";
import "./App.css";
import About from "./components/About";

import { NavBar } from "./components/NavBar";
import Works from "./components/Works";
import SkillsAndInterests from "./components/SkillsAndInterests";
import { createTheme, ThemeProvider } from "@material-ui/core/styles";
import CssBaseline from "@material-ui/core/CssBaseline";

function App() {
  const [themeFlagState, setthemeFlagState] = React.useState({
    themeFlag: true,
  });

  const theme = React.useMemo(
    () =>
      createTheme({
        palette: {
          type: themeFlagState.themeFlag ? "dark" : "light",
        },
      }),
    [themeFlagState]
  );
  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      <div className="App">
        <NavBar themeFlagState={themeFlagState} setState={setthemeFlagState} />
        <About />
        <SkillsAndInterests />
        <Works />
      </div>
    </ThemeProvider>
  );
}

export default App;
