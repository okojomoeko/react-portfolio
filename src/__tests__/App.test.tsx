import { render, screen } from "@testing-library/react";
import App from "../App";

beforeEach(() => {
  // setup a DOM element as a render target
});

afterEach(() => {
  // cleanup on exiting
});

test("renders learn react link", () => {
  const { container, getAllByRole } = render(<App />);
  const buttons = getAllByRole("button");

  const headerButtons = buttons.slice(0, 3);

  const expectHeadderButtons = ["ABOUT", "SKILLS", "WORKS"];

  let count = 0;
  for (let button of headerButtons) {
    expect(button.outerHTML).toContain(expectHeadderButtons[count]);
    count++;
  }
  expect(container.innerHTML).toMatchSnapshot();
});
