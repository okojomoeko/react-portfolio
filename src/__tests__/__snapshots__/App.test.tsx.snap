// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `"<div class=\\"App\\"><header class=\\"MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary makeStyles-appBar-2 MuiPaper-elevation4\\"><div class=\\"MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters\\"><a><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">ABOUT</span><span class=\\"MuiTouchRipple-root\\"></span></button></a><a><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">SKILLS &amp; INTEREST</span><span class=\\"MuiTouchRipple-root\\"></span></button></a><a><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">WORKS</span><span class=\\"MuiTouchRipple-root\\"></span></button></a></div><label class=\\"MuiFormControlLabel-root\\"><span class=\\"MuiSwitch-root\\"><span class=\\"MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-checked-6 Mui-checked\\" aria-disabled=\\"false\\"><span class=\\"MuiIconButton-label\\"><input class=\\"PrivateSwitchBase-input-8 MuiSwitch-input\\" name=\\"themeFlag\\" type=\\"checkbox\\" aria-label=\\"secondary checkbox\\" value=\\"\\" checked=\\"\\"><span class=\\"MuiSwitch-thumb\\"></span></span><span class=\\"MuiTouchRipple-root\\"></span></span><span class=\\"MuiSwitch-track\\"></span></span><span class=\\"MuiTypography-root MuiFormControlLabel-label MuiTypography-body1\\">Dark Theme</span></label></header><div class=\\"MuiBox-root MuiBox-root-9\\"><div class=\\"MuiBox-root MuiBox-root-10\\" id=\\"About\\"><h3 class=\\"MuiTypography-root MuiTypography-h3\\">About</h3></div>簡単な経歴<div class=\\"MuiBox-root MuiBox-root-11\\"><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-alignLeft\\"></p></div></div><div class=\\"MuiBox-root MuiBox-root-12\\"><ul><li><div class=\\"MuiBox-root MuiBox-root-18 makeStyles-oneLine-13\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3\\"><div class=\\"MuiGrid-root makeStyles-oneLineContentImg-15 MuiGrid-item MuiGrid-grid-xs-4\\"><div class=\\"makeStyles-circle-17\\"></div></div><div class=\\"MuiGrid-root makeStyles-oneLineContent-14 MuiGrid-item MuiGrid-grid-xs-8\\"><h5 class=\\"MuiTypography-root MuiTypography-h5\\"><b>2012-2016</b></h5><h5 class=\\"MuiTypography-root MuiTypography-h5\\"><b>某大学で情報科学勉強</b></h5>ちょっとだけOpenCV触ったりした</div></div></div></li><li><div class=\\"MuiBox-root MuiBox-root-19 makeStyles-oneLine-13\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3\\"><div class=\\"MuiGrid-root makeStyles-oneLineContentImg-15 MuiGrid-item MuiGrid-grid-xs-4\\"><div class=\\"makeStyles-circle-17\\"></div></div><div class=\\"MuiGrid-root makeStyles-oneLineContent-14 MuiGrid-item MuiGrid-grid-xs-8\\"><h5 class=\\"MuiTypography-root MuiTypography-h5\\"><b>2016-2018</b></h5><h5 class=\\"MuiTypography-root MuiTypography-h5\\"><b>某大学院で画像認識の勉強</b></h5>ちょっとだけ機械学習とMATLABを勉強した</div></div></div></li><li><div class=\\"MuiBox-root MuiBox-root-20 makeStyles-oneLine-13\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3\\"><div class=\\"MuiGrid-root makeStyles-oneLineContentImg-15 MuiGrid-item MuiGrid-grid-xs-4\\"><div class=\\"makeStyles-circle-17\\"></div></div><div class=\\"MuiGrid-root makeStyles-oneLineContent-14 MuiGrid-item MuiGrid-grid-xs-8\\"><h5 class=\\"MuiTypography-root MuiTypography-h5\\"><b>2018-2020</b></h5><h5 class=\\"MuiTypography-root MuiTypography-h5\\"><b>某電機メーカーに入社</b></h5>ネットワークシステムいじり。Ciscoのルータやスイッチなど触ったりネットワークテストしたり、上流工程の開発とシステム展開した</div></div></div></li><li><div class=\\"MuiBox-root MuiBox-root-21 makeStyles-oneLine-13\\"><div class=\\"MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3\\"><div class=\\"MuiGrid-root makeStyles-oneLineContentImg-15 MuiGrid-item MuiGrid-grid-xs-4\\"><div class=\\"makeStyles-circle-17\\"></div></div><div class=\\"MuiGrid-root makeStyles-oneLineContent-14 MuiGrid-item MuiGrid-grid-xs-8\\"><h5 class=\\"MuiTypography-root MuiTypography-h5\\"><b>2020.5~</b></h5><h5 class=\\"MuiTypography-root MuiTypography-h5\\"><b>転職</b></h5>クラウドを用いたシステム開発に従事</div></div></div></li></ul></div><span class=\\"makeStyles-skillAndInterestArea-22\\"><div class=\\"MuiBox-root MuiBox-root-25\\"><div class=\\"MuiBox-root MuiBox-root-26\\" id=\\"Skills\\"><h3 class=\\"MuiTypography-root MuiTypography-h3\\">Skills</h3></div><svg class=\\"MuiSvgIcon-root makeStyles-icon-24\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M20 18c1.1 0 1.99-.9 1.99-2L22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2H0v2h24v-2h-4zM4 6h16v10H4V6z\\"></path></svg><div class=\\"MuiBox-root MuiBox-root-27\\"><div class=\\"MuiPaper-root MuiCard-root makeStyles-root-23 MuiPaper-elevation1 MuiPaper-rounded\\"><div class=\\"MuiCardContent-root\\"><h6 class=\\"MuiTypography-root MuiTypography-h6\\">Languages</h6><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary\\">/ C++ /  Java /  Python /  MATLAB /  javascript /  typescript / </p></div></div></div><div class=\\"MuiBox-root MuiBox-root-28\\"><div class=\\"MuiPaper-root MuiCard-root makeStyles-root-23 MuiPaper-elevation1 MuiPaper-rounded\\"><div class=\\"MuiCardContent-root\\"><h6 class=\\"MuiTypography-root MuiTypography-h6\\">Technology</h6><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary\\">/ OpenCV /  Docker /  Node.js /  GraphQL /  Cisco / </p></div></div></div><div class=\\"MuiBox-root MuiBox-root-29\\"><div class=\\"MuiPaper-root MuiCard-root makeStyles-root-23 MuiPaper-elevation1 MuiPaper-rounded\\"><div class=\\"MuiCardContent-root\\"><h6 class=\\"MuiTypography-root MuiTypography-h6\\">Qualification</h6><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary\\">/ AWS Solutions Architect Associate /  G検定 /  E資格 / </p></div></div></div></div><div class=\\"MuiBox-root MuiBox-root-32\\"><div class=\\"MuiBox-root MuiBox-root-33\\" id=\\"Interests\\"><h3 class=\\"MuiTypography-root MuiTypography-h3\\">Interests</h3></div><svg class=\\"MuiSvgIcon-root makeStyles-icon-31\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z\\"></path></svg><div class=\\"MuiBox-root MuiBox-root-34\\"><div class=\\"MuiPaper-root MuiCard-root makeStyles-root-30 MuiPaper-elevation1 MuiPaper-rounded\\"><div class=\\"MuiCardContent-root\\"><h6 class=\\"MuiTypography-root MuiTypography-h6\\">大好き</h6><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary\\">/ スマホ /  映画 /  アニメ /  ニチアサ /  スイーツ / </p></div></div></div><div class=\\"MuiBox-root MuiBox-root-35\\"><div class=\\"MuiPaper-root MuiCard-root makeStyles-root-30 MuiPaper-elevation1 MuiPaper-rounded\\"><div class=\\"MuiCardContent-root\\"><h6 class=\\"MuiTypography-root MuiTypography-h6\\">まぁ好き</h6><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary\\">/ 情報技術 /  環境整備 / </p></div></div></div></div></span><div class=\\"MuiBox-root MuiBox-root-36\\"><div class=\\"MuiBox-root MuiBox-root-37\\" id=\\"Works\\"><h3 class=\\"MuiTypography-root MuiTypography-h3\\">Works</h3></div><div class=\\"MuiBox-root MuiBox-root-41\\"><ul class=\\"MuiList-root makeStyles-root-38 MuiList-padding\\"><li class=\\"MuiListItem-root MuiListItem-gutters\\"><a class=\\"MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\"><div class=\\"MuiListItemAvatar-root\\"><div class=\\"MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault\\"><img src=\\"/assets/img/disney-congestion.jpg\\" alt=\\"海の写真\\" title=\\"空と海\\" width=\\"100%\\" height=\\"auto\\"></div></div><div class=\\"MuiListItemText-root MuiListItemText-multiline\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">ディズニーリアルタイム待ち時間LINE-bot</span><p class=\\"MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock\\">ディズニーランド、ディズニーシーのリアルタイム待ち時間を教えてくれるLINE-bot</p></div><span class=\\"MuiTouchRipple-root\\"></span></a></li><li class=\\"MuiListItem-root MuiListItem-gutters\\"><a class=\\"MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\"><div class=\\"MuiListItemAvatar-root\\"><div class=\\"MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault\\"><img src=\\"/assets/img/infra-register.jpg\\" alt=\\"海の写真\\" title=\\"空と海\\" width=\\"100%\\" height=\\"auto\\"></div></div><div class=\\"MuiListItemText-root MuiListItemText-multiline\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">水道光熱費登録LINE-bot</span><p class=\\"MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock\\">水道光熱費をGoogleDrive上のエクセルに登録するLINE-bot</p></div><span class=\\"MuiTouchRipple-root\\"></span></a></li><li class=\\"MuiListItem-root MuiListItem-gutters\\"><a class=\\"MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\"><div class=\\"MuiListItemAvatar-root\\"><div class=\\"MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault\\"><img src=\\"/assets/img/okojotter.jpg\\" alt=\\"海の写真\\" title=\\"空と海\\" width=\\"100%\\" height=\\"auto\\"></div></div><div class=\\"MuiListItemText-root MuiListItemText-multiline\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock\\">Okojotter</span><p class=\\"MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock\\">自分用のつぶやき専用Twitterクライアント</p></div><span class=\\"MuiTouchRipple-root\\"></span></a></li></ul></div></div></div>"`;
